x/concentrated-liquidity/swaps.go

```diff
@@ -23,28 +23,36 @@ type SwapState struct {
 	liquidity                sdk.Dec // new liquidity when swap is done
 }
 
+// CreateNewConcentratedLiquidityPool creates a new concentrated liquidity pool with the given parameters.
+// The pool is created with zero liquidity and the initial sqrt price and current tick set to zero.
+// The given token denominations are ordered to ensure that the first token is the numerator of the price, and the second token is the denominator of the price.
+// The pool is added to the pool store, and an error is returned if the operation fails.
 func (k Keeper) CreateNewConcentratedLiquidityPool(
 	ctx sdk.Context,
 	poolId uint64,
 	denom0, denom1 string,
-	currSqrtPrice sdk.Dec,
-	currTick sdk.Int,
+	tickSpacing uint64,
 ) (types.ConcentratedPoolExtension, error) {
+	// Order the initial pool denoms so that denom0 is lexicographically smaller than denom1.
 	denom0, denom1, err := types.OrderInitialPoolDenoms(denom0, denom1)
 	if err != nil {
 		return nil, err
 	}
+
+	// Create a new concentrated liquidity pool with the given parameters.
 	pool := &model.Pool{
 		// TODO: move gammtypes.NewPoolAddress(poolId) to swaproutertypes
 		Address:          gammtypes.NewPoolAddress(poolId).String(),
 		Id:               poolId,
-		CurrentSqrtPrice: currSqrtPrice,
-		CurrentTick:      currTick,
+		CurrentSqrtPrice: sdk.ZeroDec(),
```

Comment:
We should also be using `NewConcentratedLiquidityPool` constructor here as opposed initializing the pool directly

Reply:
We cant do this until we move the pool model outside of the internal package. This was supposed to be done in MsgCreatePool PR but I can do that now if you think I should

Reply:
Actually disregard, I was able to do this (thought there was going to be a circular dep but there wasn't)